[2017-12-06 14:51:40.255] [ERROR] errorLogger - ReferenceError: logger is not defined
    at d:\AyyappaApp-(login&update)\routes\creayteOrUpdateEvents.js:55:76
    at d:\AyyappaApp-(login&update)\node_modules\mongoose\lib\model.js:4046:16
    at d:\AyyappaApp-(login&update)\node_modules\mongoose\lib\services\model\applyHooks.js:175:17
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2017-12-09 11:40:26.035] [ERROR] errorLogger - Error: Can't set headers after they are sent.
    at validateHeader (_http_outgoing.js:489:11)
    at ServerResponse.setHeader (_http_outgoing.js:496:3)
    at ServerResponse.header (d:\AyyappaApp-(login&update)\node_modules\express\lib\response.js:730:10)
    at ServerResponse.send (d:\AyyappaApp-(login&update)\node_modules\express\lib\response.js:170:12)
    at ServerResponse.json (d:\AyyappaApp-(login&update)\node_modules\express\lib\response.js:256:15)
    at d:\AyyappaApp-(login&update)\routes\createOrUpdateEvents.js:47:29
    at model.Query.<anonymous> (d:\AyyappaApp-(login&update)\node_modules\mongoose\lib\model.js:4046:16)
    at d:\AyyappaApp-(login&update)\node_modules\kareem\index.js:273:21
    at d:\AyyappaApp-(login&update)\node_modules\kareem\index.js:131:16
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2017-12-09 18:21:18.716] [ERROR] errorLogger - Error: Can't set headers after they are sent.
    at validateHeader (_http_outgoing.js:489:11)
    at ServerResponse.setHeader (_http_outgoing.js:496:3)
    at ServerResponse.header (d:\AyyappaApp-(login&update)\node_modules\express\lib\response.js:730:10)
    at ServerResponse.send (d:\AyyappaApp-(login&update)\node_modules\express\lib\response.js:170:12)
    at ServerResponse.json (d:\AyyappaApp-(login&update)\node_modules\express\lib\response.js:256:15)
    at d:\AyyappaApp-(login&update)\routes\createOrUpdateEvents.js:95:33
    at d:\AyyappaApp-(login&update)\node_modules\mongoose\lib\model.js:4046:16
    at d:\AyyappaApp-(login&update)\node_modules\mongoose\lib\services\model\applyHooks.js:175:17
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2017-12-16 18:38:52.677] [ERROR] errorLogger - ReferenceError: userProfileModel is not defined
    at d:\15-12 Ayyappa app\AyyappaApp\routes\createOrUpdateGroups.js:98:21
    at Query.Model.$wrapCallback (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:4046:16)
    at d:\15-12 Ayyappa app\AyyappaApp\node_modules\kareem\index.js:273:21
    at d:\15-12 Ayyappa app\AyyappaApp\node_modules\kareem\index.js:131:16
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)
[2017-12-16 18:39:44.644] [ERROR] errorLogger - ReferenceError: groupResponse is not defined
    at getUsers (d:\15-12 Ayyappa app\AyyappaApp\routes\createOrUpdateGroups.js:184:5)
    at d:\15-12 Ayyappa app\AyyappaApp\routes\createOrUpdateGroups.js:176:17
    at Model.$wrapCallback (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:4046:16)
    at next (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3172:5)
    at populate (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3277:12)
    at _populate (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3162:5)
    at Function.Model.populate (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3122:5)
    at model.populate (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\document.js:2500:19)
    at d:\15-12 Ayyappa app\AyyappaApp\routes\createOrUpdateGroups.js:167:14
    at Model.$wrapCallback (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:4046:16)
    at d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\services\model\applyHooks.js:170:20
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)
[2017-12-18 12:26:51.557] [ERROR] errorLogger - TypeError: Cannot read property 'mobileNumber' of undefined
    at getUsers (d:\15-12 Ayyappa app\AyyappaApp\routes\createOrUpdateGroups.js:195:60)
    at d:\15-12 Ayyappa app\AyyappaApp\routes\createOrUpdateGroups.js:177:17
    at Model.$wrapCallback (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:4046:16)
    at next (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3172:5)
    at populate (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3277:12)
    at _populate (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3162:5)
    at Function.Model.populate (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3122:5)
    at model.populate (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\document.js:2500:19)
    at d:\15-12 Ayyappa app\AyyappaApp\routes\createOrUpdateGroups.js:168:14
    at Model.$wrapCallback (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:4046:16)
    at d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\services\model\applyHooks.js:170:20
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)
[2017-12-18 12:29:50.178] [ERROR] errorLogger - TypeError: Cannot read property 'length' of undefined
    at getUsers (D:\15-12 Ayyappa app\AyyappaApp\routes\createOrUpdateGroups.js:200:37)
    at D:\15-12 Ayyappa app\AyyappaApp\routes\createOrUpdateGroups.js:177:17
    at D:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:4046:16
    at next (D:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3172:5)
    at populate (D:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3277:12)
    at _populate (D:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3162:5)
    at Function.Model.populate (D:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3122:5)
    at model.populate (D:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\document.js:2500:19)
    at D:\15-12 Ayyappa app\AyyappaApp\routes\createOrUpdateGroups.js:168:14
    at D:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:4046:16
    at D:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\services\model\applyHooks.js:170:20
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)
[2017-12-18 15:52:36.636] [ERROR] errorLogger - ReferenceError: object is not defined
    at d:\15-12 Ayyappa app\AyyappaApp\routes\gSearch.js:26:41
    at d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\query.js:3115:18
    at newTickHandler (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mpromise\lib\promise.js:234:18)
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)
[2017-12-18 18:51:42.788] [ERROR] errorLogger - TypeError: Cannot read property 'fullName' of undefined
    at d:\15-12 Ayyappa app\AyyappaApp\routes\createOrUpdateGroups.js:37:63
    at d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:4046:16
    at next (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3172:5)
    at populate (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3277:12)
    at _populate (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3162:5)
    at Function.Model.populate (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3122:5)
    at model.populate (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\document.js:2500:19)
    at d:\15-12 Ayyappa app\AyyappaApp\routes\createOrUpdateGroups.js:27:31
    at Query.<anonymous> (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:4046:16)
    at d:\15-12 Ayyappa app\AyyappaApp\node_modules\kareem\index.js:273:21
    at d:\15-12 Ayyappa app\AyyappaApp\node_modules\kareem\index.js:131:16
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)
[2017-12-18 19:26:47.365] [ERROR] errorLogger - TypeError: Cannot read property 'mobileNumber' of undefined
    at d:\15-12 Ayyappa app\AyyappaApp\routes\createOrUpdateGroups.js:41:70
    at d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:4046:16
    at next (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3172:5)
    at populate (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3277:12)
    at _populate (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3162:5)
    at Function.Model.populate (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:3122:5)
    at model.populate (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\document.js:2500:19)
    at d:\15-12 Ayyappa app\AyyappaApp\routes\createOrUpdateGroups.js:31:31
    at Query.<anonymous> (d:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:4046:16)
    at d:\15-12 Ayyappa app\AyyappaApp\node_modules\kareem\index.js:273:21
    at d:\15-12 Ayyappa app\AyyappaApp\node_modules\kareem\index.js:131:16
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)
[2017-12-20 17:41:43.059] [ERROR] errorLogger - TypeError: Cannot read property 'replace' of undefined
    at D:\15-12 Ayyappa app\AyyappaApp\routes\createOrUpdateGroups.js:26:58
    at Query.<anonymous> (D:\15-12 Ayyappa app\AyyappaApp\node_modules\mongoose\lib\model.js:4046:16)
    at D:\15-12 Ayyappa app\AyyappaApp\node_modules\kareem\index.js:273:21
    at D:\15-12 Ayyappa app\AyyappaApp\node_modules\kareem\index.js:131:16
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)
[2018-01-03 12:18:32.033] [ERROR] errorLogger - TypeError: Cannot read property 'indexOf' of undefined
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\createOrUpdateGroups.js:101:53
    at model.Query.<anonymous> (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:4046:16)
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:273:21
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:131:16
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-03 13:34:26.849] [ERROR] errorLogger - TypeError: Cannot read property 'indexOf' of undefined
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\createOrUpdateGroups.js:101:53
    at model.Query.<anonymous> (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:4046:16)
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:273:21
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:131:16
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-03 13:39:25.852] [ERROR] errorLogger - TypeError: Cannot read property 'indexOf' of undefined
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\createOrUpdateGroups.js:101:53
    at model.Query.<anonymous> (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:4046:16)
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:273:21
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:131:16
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-03 17:12:38.376] [ERROR] errorLogger - TypeError: groupData.populate is not a function
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\updateGroups.js:27:31
    at model.Query.<anonymous> (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:4046:16)
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:273:21
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:131:16
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-03 17:14:41.556] [ERROR] errorLogger - TypeError: groupData.populate is not a function
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\updateGroups.js:27:31
    at model.Query.<anonymous> (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:4046:16)
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:273:21
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:131:16
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-05 15:59:29.991] [ERROR] errorLogger - ReferenceError: req is not defined
    at E:\ayyapa real code\AyyappaApp_Bapatla\config\passport.js:101:17
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-09 16:53:32.153] [ERROR] errorLogger - ReferenceError: i is not defined
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\gSearch.js:25:22
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\query.js:3115:18
    at newTickHandler (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mpromise\lib\promise.js:234:18)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-09 16:56:59.198] [ERROR] errorLogger - ReferenceError: searchDataObj is not defined
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\gSearch.js:26:35
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\query.js:3115:18
    at newTickHandler (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mpromise\lib\promise.js:234:18)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-09 16:57:45.421] [ERROR] errorLogger - ReferenceError: searchDataObj is not defined
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\gSearch.js:26:35
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\query.js:3115:18
    at newTickHandler (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mpromise\lib\promise.js:234:18)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-09 17:01:49.293] [ERROR] errorLogger - ReferenceError: i is not defined
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\gSearch.js:25:22
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\query.js:3115:18
    at newTickHandler (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mpromise\lib\promise.js:234:18)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-09 17:03:26.349] [ERROR] errorLogger - ReferenceError: searchDataObj is not defined
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\gSearch.js:26:35
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\query.js:3115:18
    at newTickHandler (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mpromise\lib\promise.js:234:18)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-19 12:18:54.052] [ERROR] errorLogger - TypeError: Cannot read property '0' of undefined
    at createNewUser (E:\ayyapa real code\AyyappaApp_Bapatla\routes\userLogin.js:128:65)
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\userLogin.js:101:13
    at model.Query.<anonymous> (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:4046:16)
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:273:21
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:131:16
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-19 12:20:23.760] [ERROR] errorLogger - TypeError: Cannot read property '0' of undefined
    at createNewUser (E:\ayyapa real code\AyyappaApp_Bapatla\routes\userLogin.js:128:65)
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\userLogin.js:101:13
    at model.Query.<anonymous> (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:4046:16)
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:273:21
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:131:16
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-19 12:22:36.262] [ERROR] errorLogger - TypeError: Cannot read property 'name' of undefined
    at createNewUser (E:\ayyapa real code\AyyappaApp_Bapatla\routes\userLogin.js:128:66)
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\userLogin.js:101:13
    at model.Query.<anonymous> (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:4046:16)
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:273:21
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:131:16
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-19 17:26:30.419] [ERROR] errorLogger - ReferenceError: u is not defined
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\createOrUpdateGroups.js:115:36
    at model.Query.<anonymous> (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:4046:16)
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:273:21
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:131:16
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-19 19:36:11.943] [ERROR] errorLogger - TypeError: Cannot read property 'username' of undefined
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\createOrUpdateEvents.js:71:70
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:4046:16
    at next (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:3172:5)
    at populate (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:3277:12)
    at _populate (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:3162:5)
    at Function.Model.populate (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:3122:5)
    at model.populate (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\document.js:2500:19)
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\createOrUpdateEvents.js:69:38
    at model.Query.<anonymous> (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:4046:16)
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:273:21
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:131:16
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-19 21:02:41.738] [ERROR] errorLogger - TypeError: Cannot read property 'username' of undefined
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\createOrUpdateEvents.js:84:76
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:4046:16
    at next (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:3172:5)
    at populate (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:3277:12)
    at _populate (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:3162:5)
    at Function.Model.populate (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:3122:5)
    at model.populate (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\document.js:2500:19)
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\createOrUpdateEvents.js:82:38
    at model.Query.<anonymous> (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:4046:16)
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:273:21
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:131:16
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-19 21:41:52.382] [ERROR] errorLogger - ReferenceError: gdata is not defined
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\createOrUpdateEvents.js:84:29
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:4046:16
    at next (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:3172:5)
    at next (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:3289:7)
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\query.js:3115:18
    at newTickHandler (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mpromise\lib\promise.js:234:18)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-19 21:46:45.032] [ERROR] errorLogger - ReferenceError: gdata is not defined
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\createOrUpdateEvents.js:84:29
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:4046:16
    at next (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:3172:5)
    at next (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:3289:7)
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\query.js:3115:18
    at newTickHandler (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mpromise\lib\promise.js:234:18)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-20 16:46:46.999] [ERROR] errorLogger - TypeError: Cannot read property 'indexOf' of undefined
    at E:\ayyapa real code\AyyappaApp_Bapatla\routes\createOrUpdateGroups.js:90:53
    at model.Query.<anonymous> (E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\mongoose\lib\model.js:4046:16)
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:273:21
    at E:\ayyapa real code\AyyappaApp_Bapatla\node_modules\kareem\index.js:131:16
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-23 12:38:31.791] [ERROR] errorLogger - ReferenceError: port is not defined
    at Server.onError (E:\ayyapa real code\AyyappaApp_Bapatla\mongodb\Connections.js:71:21)
    at emitOne (events.js:116:13)
    at Server.emit (events.js:211:7)
    at emitErrorNT (net.js:1371:8)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-23 12:39:14.904] [ERROR] errorLogger - ReferenceError: port is not defined
    at Server.onError (E:\ayyapa real code\AyyappaApp_Bapatla\mongodb\Connections.js:71:21)
    at emitOne (events.js:116:13)
    at Server.emit (events.js:211:7)
    at emitErrorNT (net.js:1371:8)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-01-23 12:41:34.503] [ERROR] errorLogger - ReferenceError: port is not defined
    at Server.onError (E:\ayyapa real code\AyyappaApp_Bapatla\mongodb\Connections.js:71:21)
    at emitOne (events.js:116:13)
    at Server.emit (events.js:211:7)
    at emitErrorNT (net.js:1371:8)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-05-24 18:03:46.588] [ERROR] errorLogger - ReferenceError: conn is not defined
    at NativeConnection.dbEvents.once (E:\Fanzingo services code\Fanzingo_Bapatla\mongodb\Connections.js:49:14)
    at Object.onceWrapper (events.js:313:30)
    at emitNone (events.js:111:20)
    at NativeConnection.emit (events.js:208:7)
    at E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongoose\lib\connection.js:860:13
    at connectCallback (E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb\lib\mongo_client.js:515:5)
    at E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb\lib\mongo_client.js:449:13
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-05-24 18:06:51.301] [ERROR] errorLogger - ReferenceError: mongoose is not defined
    at NativeConnection.dbEvents.once (E:\Fanzingo services code\Fanzingo_Bapatla\mongodb\Connections.js:49:27)
    at Object.onceWrapper (events.js:313:30)
    at emitNone (events.js:111:20)
    at NativeConnection.emit (events.js:208:7)
    at E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongoose\lib\connection.js:860:13
    at connectCallback (E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb\lib\mongo_client.js:515:5)
    at E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb\lib\mongo_client.js:449:13
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-05-24 18:09:29.627] [ERROR] errorLogger - Error: missing mongo argument
new Grid(db, mongo)
    at new Grid (E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\gridfs-stream\lib\index.js:24:21)
    at Grid (E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\gridfs-stream\lib\index.js:19:12)
    at NativeConnection.dbEvents.once (E:\Fanzingo services code\Fanzingo_Bapatla\mongodb\Connections.js:49:9)
    at Object.onceWrapper (events.js:313:30)
    at emitNone (events.js:111:20)
    at NativeConnection.emit (events.js:208:7)
    at E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongoose\lib\connection.js:860:13
    at connectCallback (E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb\lib\mongo_client.js:515:5)
    at E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb\lib\mongo_client.js:449:13
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-05-24 18:14:06.635] [ERROR] errorLogger - Error: missing mongo argument
new Grid(db, mongo)
    at new Grid (E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\gridfs-stream\lib\index.js:24:21)
    at Grid (E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\gridfs-stream\lib\index.js:19:12)
    at NativeConnection.dbEvents.once (E:\Fanzingo services code\Fanzingo_Bapatla\mongodb\Connections.js:49:9)
    at Object.onceWrapper (events.js:313:30)
    at emitNone (events.js:111:20)
    at NativeConnection.emit (events.js:208:7)
    at E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongoose\lib\connection.js:860:13
    at connectCallback (E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb\lib\mongo_client.js:515:5)
    at E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb\lib\mongo_client.js:449:13
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-05-24 19:21:29.925] [ERROR] errorLogger - Error: missing mongo argument
new Grid(db, mongo)
    at new Grid (E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\gridfs-stream\lib\index.js:24:21)
    at Grid (E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\gridfs-stream\lib\index.js:19:12)
    at NativeConnection.dbEvents.once (E:\Fanzingo services code\Fanzingo_Bapatla\mongodb\Connections.js:49:9)
    at Object.onceWrapper (events.js:313:30)
    at emitNone (events.js:111:20)
    at NativeConnection.emit (events.js:208:7)
    at E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongoose\lib\connection.js:860:13
    at connectCallback (E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb\lib\mongo_client.js:515:5)
    at E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb\lib\mongo_client.js:449:13
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-05-24 19:23:10.233] [ERROR] errorLogger - Error: missing mongo argument
new Grid(db, mongo)
    at new Grid (E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\gridfs-stream\lib\index.js:24:21)
    at Grid (E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\gridfs-stream\lib\index.js:19:12)
    at NativeConnection.dbEvents.once (E:\Fanzingo services code\Fanzingo_Bapatla\mongodb\Connections.js:50:9)
    at Object.onceWrapper (events.js:313:30)
    at emitNone (events.js:111:20)
    at NativeConnection.emit (events.js:208:7)
    at E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongoose\lib\connection.js:860:13
    at connectCallback (E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb\lib\mongo_client.js:515:5)
    at E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb\lib\mongo_client.js:449:13
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
[2018-05-25 11:01:58.758] [ERROR] errorLogger - ReferenceError: Grid is not defined
    at NativeConnection.conn.once (E:\Fanzingo services code\Fanzingo_Bapatla\routes\capitalAlphabets.js:26:3)
    at Object.onceWrapper (events.js:313:30)
    at emitNone (events.js:106:13)
    at NativeConnection.emit (events.js:208:7)
    at open (E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongoose\lib\connection.js:738:11)
    at E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongoose\lib\connection.js:744:7
    at E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb\lib\authenticate.js:96:5
    at handleCallback (E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb\lib\utils.js:120:56)
    at _callback (E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb\lib\authenticate.js:23:5)
    at E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb\lib\authenticate.js:66:7
    at E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb-core\lib\connection\pool.js:760:7
    at E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb-core\lib\connection\pool.js:738:11
    at finish (E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb-core\lib\auth\scram.js:165:16)
    at handleEnd (E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb-core\lib\auth\scram.js:178:7)
    at E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb-core\lib\auth\scram.js:266:13
    at E:\Fanzingo services code\Fanzingo_Bapatla\node_modules\mongodb-core\lib\connection\pool.js:469:18
[2019-06-28 12:21:15.785] [ERROR] errorLogger - TypeError: Cannot read property '_id' of undefined
    at D:\Fanzingo services code\ChatDemoApp\routes\userLogin.js:50:42
    at D:\Fanzingo services code\ChatDemoApp\node_modules\mongoose\lib\model.js:4046:16
    at D:\Fanzingo services code\ChatDemoApp\node_modules\mongoose\lib\services\model\applyHooks.js:175:17
    at process._tickCallback (internal/process/next_tick.js:61:11)
[2019-06-28 12:23:35.358] [ERROR] errorLogger - TypeError: Cannot read property '_id' of undefined
    at D:\Fanzingo services code\ChatDemoApp\routes\userLogin.js:51:42
    at D:\Fanzingo services code\ChatDemoApp\node_modules\mongoose\lib\model.js:4046:16
    at D:\Fanzingo services code\ChatDemoApp\node_modules\mongoose\lib\services\model\applyHooks.js:175:17
    at process._tickCallback (internal/process/next_tick.js:61:11)
[2019-06-28 12:27:42.547] [ERROR] errorLogger - TypeError: Cannot read property '_id' of undefined
    at D:\Fanzingo services code\ChatDemoApp\routes\userLogin.js:51:42
    at D:\Fanzingo services code\ChatDemoApp\node_modules\mongoose\lib\model.js:4046:16
    at D:\Fanzingo services code\ChatDemoApp\node_modules\mongoose\lib\services\model\applyHooks.js:175:17
    at process._tickCallback (internal/process/next_tick.js:61:11)
[2019-06-28 12:34:55.669] [ERROR] errorLogger - TypeError: Cannot read property '_id' of undefined
    at D:\Fanzingo services code\ChatDemoApp\routes\userLogin.js:51:42
    at D:\Fanzingo services code\ChatDemoApp\node_modules\mongoose\lib\model.js:4046:16
    at D:\Fanzingo services code\ChatDemoApp\node_modules\mongoose\lib\services\model\applyHooks.js:175:17
    at process._tickCallback (internal/process/next_tick.js:61:11)
[2019-06-28 13:39:22.337] [ERROR] errorLogger - TypeError: Cannot read property '_id' of undefined
    at D:\Fanzingo services code\ChatDemoApp\routes\userLogin.js:50:42
    at D:\Fanzingo services code\ChatDemoApp\node_modules\mongoose\lib\model.js:4046:16
    at D:\Fanzingo services code\ChatDemoApp\node_modules\mongoose\lib\services\model\applyHooks.js:175:17
    at process._tickCallback (internal/process/next_tick.js:61:11)
[2019-06-28 14:38:52.130] [ERROR] errorLogger - ReferenceError: response is not defined
    at D:\Fanzingo services code\ChatDemoApp\routes\messages.js:21:17
    at D:\Fanzingo services code\ChatDemoApp\node_modules\mongoose\lib\query.js:3115:18
    at newTickHandler (D:\Fanzingo services code\ChatDemoApp\node_modules\mpromise\lib\promise.js:234:18)
    at process._tickCallback (internal/process/next_tick.js:61:11)
[2019-06-28 14:41:40.173] [ERROR] errorLogger - ReferenceError: response is not defined
    at D:\Fanzingo services code\ChatDemoApp\routes\messages.js:21:17
    at D:\Fanzingo services code\ChatDemoApp\node_modules\mongoose\lib\query.js:3115:18
    at newTickHandler (D:\Fanzingo services code\ChatDemoApp\node_modules\mpromise\lib\promise.js:234:18)
    at process._tickCallback (internal/process/next_tick.js:61:11)
[2019-06-28 15:32:24.160] [ERROR] errorLogger - ReferenceError: response is not defined
    at D:\Fanzingo services code\ChatDemoApp\routes\messages.js:21:17
    at D:\Fanzingo services code\ChatDemoApp\node_modules\mongoose\lib\query.js:3115:18
    at newTickHandler (D:\Fanzingo services code\ChatDemoApp\node_modules\mpromise\lib\promise.js:234:18)
    at process._tickCallback (internal/process/next_tick.js:61:11)
[2019-06-28 18:16:29.217] [ERROR] errorLogger - ReferenceError: feeds is not defined
    at D:\Fanzingo services code\ChatDemoApp\routes\getMessages.js:32:37
    at D:\Fanzingo services code\ChatDemoApp\node_modules\mongoose\lib\aggregate.js:767:13
    at D:\Fanzingo services code\ChatDemoApp\node_modules\kareem\index.js:131:16
    at process._tickCallback (internal/process/next_tick.js:61:11)
[2020-10-19 19:24:50.427] [ERROR] errorLogger - TypeError: Cannot read property '_id' of undefined
    at D:\Algimetric Services\routes\userLogin.js:50:42
    at D:\Algimetric Services\node_modules\mongoose\lib\model.js:4046:16
    at D:\Algimetric Services\node_modules\mongoose\lib\services\model\applyHooks.js:175:17
    at processTicksAndRejections (internal/process/task_queues.js:79:11)
[2020-10-19 19:27:53.194] [ERROR] errorLogger - TypeError: Cannot read property '_id' of undefined
    at D:\Algimetric Services\routes\userLogin.js:51:42
    at D:\Algimetric Services\node_modules\mongoose\lib\model.js:4046:16
    at D:\Algimetric Services\node_modules\mongoose\lib\services\model\applyHooks.js:175:17
    at processTicksAndRejections (internal/process/task_queues.js:79:11)
[2020-10-20 08:08:12.551] [ERROR] errorLogger - ReferenceError: messages is not defined
    at D:\Algimetric Services\routes\userLogin.js:31:23
    at model.Query.<anonymous> (D:\Algimetric Services\node_modules\mongoose\lib\model.js:4046:16)
    at D:\Algimetric Services\node_modules\kareem\index.js:273:21
    at D:\Algimetric Services\node_modules\kareem\index.js:131:16
    at processTicksAndRejections (internal/process/task_queues.js:79:11)
